elytron/WFCORE-3596-named-permission-sets.adoc

```diff
@@ -0,0 +1,108 @@
+  Notice that the default permissions will be in a default permission set, "default-permissions", that is referenced by
+  the "default-permission-mapper". For the provisioning tool, the default permission set will be the single place to add
+  permissions as subsystems are being added and it will be the single place to remove permissions as subsystems are being
+  removed. Without named permission sets, the provisioning tool would not know which permission mapping a subsystem
+  permission should be added to or removed from. Note that the provisioning tool will only be touching the "default-permissions".
+  It won't attempt to touch any user-specified permission sets.
+
+[source,xml]
+----
+<mappers>
+...
+    <simple-permission-mapper name="default-permission-mapper" mapping-mode="first">
+        <permission-mapping>
+            <principal name="anonymous"/>
+            <permission-set name="default-permissions"/>
+        </permission-mapping>
+        <permission-mapping match-all="true">
+            <permission-set name="login-permission" />
+            <permission-set name="default-permissions"/>
+        </permission-mapping>
+    </simple-permission-mapper>
+...
+</mappers>
+
+<permission-sets>
+    <permission-set name="login-permission">
+        <permission class-name="org.wildfly.security.auth.permission.LoginPermission"/>
+    </permission-set>
+    <permission-set name="default-permissions">
+        <permission class-name="org.wildfly.extension.batch.jberet.deployment.BatchPermission" module="org.wildfly.extension.batch.jberet" target-name="*"/>
+        <permission class-name="org.wildfly.transaction.client.RemoteTransactionPermission" module="org.wildfly.transaction.client"/>
+        <permission class-name="org.jboss.ejb.client.RemoteEJBPermission" module="org.jboss.ejb-client"/>
+    </permission-set>
+</permission-sets>
+----
+
+* A permission set should be a private capability.
+
+* When transforming the model for legacy slaves, the permission sets should be in-lined, i.e., a permission set that's
+  referenced from a permission mapping or from a constant permission mapper would just be replaced with the permissions
+  that make up the permission set. For example, the "default-permissions" permission set referenced from the permission
+  mappings in the above example would be transformed to:
+
+[source,xml]
+----
+<permission class-name="org.wildfly.extension.batch.jberet.deployment.BatchPermission" module="org.wildfly.extension.batch.jberet" target-name="*"/>
+<permission class-name="org.wildfly.transaction.client.RemoteTransactionPermission" module="org.wildfly.transaction.client"/>
+<permission class-name="org.jboss.ejb.client.RemoteEJBPermission" module="org.jboss.ejb-client"/>
+----
+
+* XML configuration for prior versions should be converted to the new version, i.e., each list of permissions should be
```

Comment:
Is this meant manually? If so, this is compatibility issue. I think user should be able to copy standalone.xml to newer version of server without issue.

Reply:
This will be done automatically.